test/e2e_node/shortened_grace_period_test.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"github.com/onsi/ginkgo/v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+	watchtools "k8s.io/client-go/tools/watch"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	admissionapi "k8s.io/pod-security-admission/api"
+	"strings"
+	"time"
+)
+
+var _ = SIGDescribe("Shortened Grace Period", func() {
+	f := framework.NewDefaultFramework("shortened-grace-period")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelBaseline
+	ginkgo.Context("When repeatedly deleting pods", func() {
+		var podClient *e2epod.PodClient
+		var dc dynamic.Interface
+		var ns string
+		var podName = "test"
+		var ctx = context.Background()
+		var rcResource = schema.GroupVersionResource{Group: "", Version: "v1", Resource: "Pod"}
+		ginkgo.BeforeEach(func() {
+			ns = f.Namespace.Name
+			dc = f.DynamicClient
+			podClient = e2epod.NewPodClient(f)
+		})
+		ginkgo.It("shorter grace period of a second command overrides the longer grace period of a first command", func() {
+			expectedWatchEvents := []watch.Event{
+				{Type: watch.Deleted},
+			}
+			var exitCode int32
+			callback := func(retryWatcher *watchtools.RetryWatcher) (actualWatchEvents []watch.Event) {
+				pod, err := podClient.Get(ctx, podName, metav1.GetOptions{})
+				framework.ExpectNoError(err, "failed to get pod %q", podName)
+				// Verify exit code
+				exitCode = pod.Status.ContainerStatuses[0].State.Terminated.ExitCode
+				framework.ExpectNoError(err, "failed to get most recent container exit code for pod %q", podName)
```

Comment:
This check doesn't look right. We're already checking the error code once above

Reply:
Do you think this place is useless? Do you need to delete this section?